src/recipe/Baked.Recipe.Service.Application/Theme/Admin/Components.cs

```diff
@@ -33,8 +33,17 @@ public static ComponentDescriptorAttribute<DataPanel> DataPanel(string title, IC
 
     public static ComponentDescriptorAttribute<DataPanel> DataPanel(IData title, IComponentDescriptor content,
         IEnumerable<Parameter>? parameters = default,
+        bool? localizeTitle = default,
         bool? collapsed = default
-    ) => new(new(title, content) { Collapsed = collapsed, Parameters = [.. parameters ?? []] });
+    ) => new(new(title, content)
+    {
+        Collapsed = collapsed,
+        LocalizeTitle =
+            localizeTitle is not null ? localizeTitle :
+            title is InlineData ? true :
+            null,
```

Comment:
let's provide a `RequiresLocalization` property from `IData` interface, and instead of checking if data is inline data.

make sure `RequireLocalization` is not serialized to descriptors, i assume explicity implementations wouldn't serialize.

all data implementations can have this option. so we can remove `localizeTitle` parameter from component helper methods. and set `LocalizeTitle` property from `RequiresLocalization` property directly.